Mathlib/Data/List/Basic.lean

```diff
@@ -2278,8 +2276,7 @@ theorem disjoint_reverse_right {l₁ l₂ : List α} : Disjoint l₁ l₂.revers
 end Disjoint
 
 section lookup
-
-variable {α β : Type*} [BEq α] [LawfulBEq α]
+variable [BEq α] [LawfulBEq α]
```

Comment:
The removal of alpha here is surprising to me. Is this change secretly changing the implicitness of alpha below? I don't really see how else the old and new version could both compile

Reply:
They were already defined and already implicit, so this was redundant here.

Reply:
Huh, I thought that `variables` which were defined and not changing explicitness gave an error. Thanks for explaining!

Reply:
To be completely explicit: a *binder update* gives an error, but *shadowing* does not:
```
variable {α : Type}
variable {α : Type} -- no problem!
variable {α} -- redundant binder annotation update
```
This command was not a binder update, it was shadowing variables defined at the beginning of the file.

Reply:
Ah, that is indeed exactly the confusion I had. Thanks again!